plugins/languages/treesitter/treesitter.nix

```diff
@@ -6,210 +6,308 @@
   ...
 }:
 with lib;
-let
-  cfg = config.plugins.treesitter;
-in
-{
-  options = {
-    plugins.treesitter = {
-      enable = mkEnableOption "tree-sitter syntax highlighting";
-
-      package = mkOption {
-        type = types.package;
-        default = pkgs.vimPlugins.nvim-treesitter;
-        description = "Plugin to use for nvim-treesitter. If using nixGrammars, it should include a `withPlugins` function";
-      };
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "treesitter";
+  originalName = "nvim-treesitter";
+  luaName = "nvim-treesitter.configs";
+  defaultPackage = pkgs.vimPlugins.nvim-treesitter;
 
-      nixGrammars = mkOption {
-        type = types.bool;
-        default = true;
-        description = "Install grammars with Nix";
-      };
+  maintainers = [ lib.maintainers.khaneliman ];
 
-      ensureInstalled = mkOption {
-        type =
-          with types;
-          oneOf [
-            (enum [ "all" ])
-            (listOf str)
-          ];
-        default = "all";
-        description = "Either \"all\" or a list of languages";
-      };
+  # TODO introduced 2024-07-06: remove after 24.11
+  optionsRenamedToSettings = [
+    "ensureInstalled"
+    "ignoreInstall"
+    "parserInstallDir"
+    [
+      "incrementalSelection"
+      "enable"
+    ]
+    [
+      "incrementalSelection"
+      "keymaps"
+      "initSelection"
+      "nodeDecremental"
+    ]
+    [
+      "incrementalSelection"
+      "keymaps"
+      "initSelection"
+      "nodeIncremental"
+    ]
+    [
+      "incrementalSelection"
+      "keymaps"
+      "initSelection"
+      "scopeIncremental"
+    ]
+  ];
 
-      gccPackage = helpers.mkPackageOption {
-        default = if cfg.nixGrammars then null else pkgs.gcc;
-        description = ''
-          Which package (if any) to be added as the GCC compiler.
-          This is required to build grammars if you are not using `nixGrammars`.
-          To disable the installation of GCC, set this option to `null`.
-        '';
-      };
+  imports =
+    let
+      basePluginPath = [
+        "plugins"
+        "treesitter"
+      ];
+      settingsPath = basePluginPath ++ [ "settings" ];
+    in
+    [
```

Comment:
```suggestion
    [
      (lib.mkRenamedOptionModule (basePluginPath ++ [ "moduleConfig" ]) settingsPath)
```

Reply:
I'm so confused where that change disappeared... i just added that lol